src/rgw/rgw_rados.cc

```diff
@@ -4183,7 +4247,7 @@ void RGWRados::build_bucket_index_marker(const string& shard_id_str, const strin
 
 int RGWRados::open_bucket_index_ctx(rgw_bucket& bucket, librados::IoCtx& index_ctx)
 {
-  int r = open_pool_ctx(bucket.index_pool, index_ctx);
+  int r = open_pool_ctx(bucket.index_pool, index_ctx, bucket.ns);
```

Comment:
@cbodley : Should the namespace be set for index pool also? I don't think we should. Please let me know what is correct.